tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs

```diff
@@ -7395,6 +7396,95 @@ await ExecuteWorkerAsync<WorkflowUsingPriorities>(
             new TemporalWorkerOptions().AddAllActivities<WorkflowUsingPriorities>(null));
     }
 
+    [Workflow]
+    public class WaitOnSignalWorkflowThenActivity
+    {
+        private bool complete;
+
+        [WorkflowRun]
+        public async Task RunAsync()
+        {
+            await Workflow.WaitConditionAsync(() => complete);
+            await Workflow.ExecuteActivityAsync(() => SomeActivity(), new()
+            {
+                StartToCloseTimeout = TimeSpan.FromSeconds(10),
+            });
+        }
+
+        [WorkflowSignal]
+        public async Task CompleteAsync() => complete = true;
+
+        [Activity]
+        public static string SomeActivity() => "activity-result";
+    }
+
+    [Fact]
+    public async Task AutoscallingPollingWorker()
+    {
+        var promAddr = $"127.0.0.1:{TestUtils.FreePort()}";
+        var runtime = new TemporalRuntime(new()
+        {
+            Telemetry = new()
+            {
+                Metrics = new() { Prometheus = new(promAddr), },
+            },
+        });
+        var client = await TemporalClient.ConnectAsync(
+            new()
+            {
+                TargetHost = Client.Connection.Options.TargetHost,
+                Namespace = Client.Options.Namespace,
+                Runtime = runtime,
+            });
+
+        await ExecuteWorkerAsync<WaitOnSignalWorkflowThenActivity>(
+        async worker =>
+        {
+            await client.WorkflowService.GetSystemInfoAsync(new());
+
+            // Give pollers a beat to start
+            await Task.Delay(TimeSpan.FromMilliseconds(300));
```

Comment:
Instead of sleeping up front for a fixed time, can you wrap what you want to assert into an `AssertMore.EventuallyAsync`?